app/services/api/map_hesa_attributes/v01.rb

```diff
@@ -20,8 +22,9 @@ def self.disability_attributes(params)
         params[:data].keys.select { |key| key.to_s.match(DISABILITY_PARAM_REGEX) }
       end
 
-      def initialize(params:)
-        @params = params
+      def initialize(trainee: nil, params:)
```

Comment:
Yeha, the mapper is now using trainee to default/retain old value, kinda overlap with `trainee_attribute_service.from_trainee(traine)`

Reply:
> Seems a bit odd to introduce a dependency on the activerecord model here because at this stage in the pipeline we are just dealing with the raw input data.
> 
> We probably need to handle the case where `params[:ethnicity]` is present and `::Hesa::CodeSets::Ethnicities::MAPPING[params[:ethnicity]]` is `nil` as a validation error. So we probably need to retain the original `params[:ethnicity]` value at this stage so that it can be validated later in the pipeline.

@stevehook the `params[:ethnicity]` even if is not included in the request body is going to end up `nil` since we use the `ActiveModel::Attributes`. So I don't think it would be possible to validate for presence as the API should still consider the request valid without the `ethnicity` in the request body.


> Yeha, the mapper is now using trainee to default/retain old value, kinda overlap with `trainee_attribute_service.from_trainee(traine)`

@defong Yes agree, the issue is that we call `trainee_attribute_service.from_trainee(traine)` before we assign the mapped attributes therefore they will get overriden.

```
attributes = trainee_attributes_service.from_trainee(trainee)
attributes.assign_attributes(hesa_mapped_params_for_update(trainee))
```

Reply:
This could work for the update, however when the Trainee is first created won't receive an `ethnic_group` = `not_provided_ethnic_group` and `ethnic_background` = `Not provided` and will result in a validation error

- [has_diversity_attributes.rb#L9](https://github.com/DFE-Digital/register-trainee-teachers/blob/7051-some-fields-should-be-read-only/app/services/concerns/has_diversity_attributes.rb#L9)
- [form_validator.rb#L21](https://github.com/DFE-Digital/register-trainee-teachers/blob/7051-some-fields-should-be-read-only/app/forms/diversities/form_validator.rb#L21)

Reply:
It looks like we might have a discrepancy between how the form validator works (because of the normal form flow) and how we want to handle ethnicities coming from the API.